### PR TITLE
Divide src/progress.rs into mods

### PR DESCRIPTION
**Pull request by Fullstop000**
https://github.com/pingcap/raft-rs/issues/125

The original `src/progress.rs` is divided into `src/progress/{mod.rs, infights.rs, progress_set.rs}`

### CODE REVIEW

**Comment by Hoverbear** [comment]
Hi @Fullstop000 !

I think this is a step in the right direction. :) In the future we may want to split things up more, and there are certainly other parts of Raft that could have this treatment, but I appreciate you choosing to do this in a bite sized chunk.

Please take a moment to fix the merge conflict if you could.

**Comment by siddontang** [comment]
PTAL @BusyJay

File: src/progress/inflights.rs [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+    pub fn new(cap: usize) -> Inflights {
+        Inflights {
+            buffer: Vec::with_capacity(cap),
+            start: 0,
+            count: 0,
+        }
+    }
+
+    /// Returns true if the inflights is full.
+    pub fn full(&self) -> bool {
```
Comment (siddontang) [inline src/progress/inflights.rs]: seem we can use inline for all simple functions.

Comment (Fullstop000) [reply src/progress/inflights.rs]: That's right. I'll do the checking.

**Comment by Hoverbear** [comment]
@BusyJay Was your last review a green approved check? :) Can you give another?

**Comment by BusyJay** [comment]
Thanks for your contribution!

**Pull request merged**